### DISCUSSION TITLE
How to setup unit tests when using shiny?

### DISCUSSION THREAD

**Discussion by lemcoder** [shinyorg/shiny#1195] [Q&A] [answered]
Hello!
I wanted to ask if there is a preffered way of setting up the test project when using shiny for MAUI. I want to test my viewModels that ate using some of the Shiny library functions.

Thank's in advance!

**Comment by aritchie** [comment, marked as answer]
You can't unit test a lot of things in Shiny like bluetooth.  You can unit test things like the jobs and background delegates.

Shiny has netstandard2.0 targets for all the interfaces that you can mock as needed.
